### PR TITLE
Test all available python3 new versions if any

### PR DESCRIPTION
**Pull request by rfan1**
https://progress.opensuse.org/issues/129478

- Verification run:
[15SP4/15SP5/TW](http://bassey.arch.nue2.suse.org/tests/overview?&build=rfan0518)
Test failure on sle15sp5 is a known issue:https://bugzilla.suse.com/show_bug.cgi?id=1211386

### CODE REVIEW

File: tests/console/python3_new_version_check.pm [hunk 1]
```diff
@@ -32,27 +32,36 @@ sub run {
...
+        my $man_or_boy = script_output("python3.$sub_version man_or_boy.py");
+        if ($man_or_boy != -67) {
+            die("Execution of 'man_or_boy.py' with $python3_version is not correct\n");
         }
     }
 }
 
+sub post_fail_hook {
+    select_console 'log-console';
+    assert_script_run 'save_y2logs /tmp/patterns_y2logs.tar.bz2';
```
Comment (dzedro) [inline tests/console/python3_new_version_check.pm:63]: patterns ? :wink:

File: tests/console/python3_new_version_check.pm [hunk 2]
```diff
@@ -32,27 +32,36 @@ sub run {
...
-            if ($man_or_boy != -67) {
-                die("Execution of 'man_or_boy.py' not correct\n");
-            }
+        die('No new python3 packages available');
+    }
+    my @python3_versions = split(/\n/, script_output(qq[zypper se 'python3[0-9]*' | awk -F '|' '/python3[0-9]/ {gsub(" ", ""); print \$2}' | awk -F '-' '{print \$1}' | uniq]));
+    record_info("All available new python3 versions are: @python3_versions");
+    my $sub_version;
+    foreach my $python3_version (@python3_versions) {
+        record_info("$python3_version is tested now");
```
Comment (dzedro) [inline tests/console/python3_new_version_check.pm:48]: Please move the longer sentence to second part, it's not visible anyway.
like "Tested python$python3_version", "$python3_version is tested now"
Not sure what will fit in the square.

Comment (dzedro) [reply tests/console/python3_new_version_check.pm:48]: or just `"Testing $python3_version", "$python3_version is tested now"`

File: tests/console/python3_new_version_check.pm [hunk 3]
```diff
@@ -32,27 +32,36 @@ sub run {
...
-    my @python_versions = split(/,/, get_var('PYTHON_VERSIONS', '3.10'));
-    my $sub_version;
-    foreach my $python_version (@python_versions) {
-        record_info("python$python_version is tested now");
-        $sub_version = substr($python_version, 2);
-        my @python31x_results = split "\n", script_output("zypper se python3$sub_version | awk -F \'|\' \'/python3$sub_version/ {gsub(\" \", \"\"); print \$2}\'");
-        if ((scalar(@python31x_results) == 0) || ($python31x_results[0] ne "python3$sub_version")) {
-            die("Python $python_version not found in the repositories\n\n");
+    # Test all avaiable new python3 versions if any
+    my $ret = zypper_call('se "python3[0-9]*"', exitcode => [0, 104]);
```
Comment (dzedro) [inline tests/console/python3_new_version_check.pm:36]: Are you sure ? https://openqa.suse.de/tests/11153979#step/python3_new_version_check/24

File: tests/console/python3_new_version_check.pm [hunk 4]
```diff
@@ -32,27 +32,36 @@ sub run {
...
+    my @python3_versions = split(/\n/, script_output(qq[zypper se 'python3[0-9]*' | awk -F '|' '/python3[0-9]/ {gsub(" ", ""); print \$2}' | awk -F '-' '{print \$1}' | uniq]));
+    record_info("Available versions", "All available new python3 versions are: @python3_versions");
+    my $sub_version;
+    foreach my $python3_version (@python3_versions) {
+        record_info("Testing $python3_version", "$python3_version is tested now");
+        $sub_version = substr($python3_version, 7);
+        zypper_call("install $python3_version");
+        # Running classic testing algorithm 'man_or_boy'. More info at:
+        # https://rosettacode.org/wiki/Man_or_boy_test
+        assert_script_run("[ -f man_or_boy.py ] || curl -O " . data_url("console/man_or_boy.py") . " || true");
```
Comment (dzedro) [inline tests/console/python3_new_version_check.pm:53]: Could this be downloaded just once before the loop ?

File: tests/console/python3_new_version_check.pm (hunk 3 above)
Comment (rfan1) [reply tests/console/python3_new_version_check.pm:36]: [SLE15SP5_QEM](http://bassey.arch.nue2.suse.org/tests/58#step/python3_new_version_check/40)

Comment (rfan1) [reply tests/console/python3_new_version_check.pm:36]: Now I can remove the soft failure, thanks much!

**Review by dzedro — APPROVED** [review]

**Pull request merged**